### ISSUE TITLE
Object.isInvalidated should be @objc dynamic for KVO

### ISSUE THREAD

**Issue by sadlerjw** [realm/realm-cocoa#6554]
TLDR: observing `Object.isInvalidated` with KVO works just fine in Swift, but produces compiler warnings in Swift 5.2. This should be able to be fixed by just adding the `@objc` and `dynamic` attributes to the `isInvalidated` property.

## Goals
Observe `isInvalidated` on my `Object` subclasses so that I can know when they are deleted or otherwise invalidated so I can dismiss any UIs that assume that they are valid. We use this to avoid crashes that happen by accessing properties on invalidated objects.

## Expected Results
Observing `isInvalidated` works and does not produce compilation warnings.

## Actual Results
Observing `isInvalidated` works but produces a compilation warning. In our project we like to stay warning-free and have "treat warnings as errors" enabled.

This used to work just fine, but as of Swift 5.2 there is a new warning if you attempt to perform KVO on a property that's not marked as both `@objc` and `dynamic`: 
```
Passing reference to non-'@objc dynamic' property 'isInvalidated' to KVO method 'observe(_:options:changeHandler:)' may lead to unexpected behavior or runtime trap
```

## Steps to Reproduce
See code sample below. Compile using Xcode 11.5 or later in Swift 5 mode.

## Code Sample
```swift
let observation = myObject?.observe(\.isInvalidated, changeHandler: { [weak self] _, _ in
    self?.handleObjectBecomingInvalidated()
})
```

## Version of Realm and Tooling
```
ProductName:	Mac OS X
ProductVersion:	10.15.5
BuildVersion:	19F101

/Applications/Xcode.app/Contents/Developer
Xcode 11.5
Build version 11E608c

/usr/local/bin/pod
1.9.1
Realm (3.17.3)
RealmSwift (3.17.3)
RealmSwift (~> 3.17.3)

/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)

/usr/local/bin/carthage
0.34.0
(not in use here)

/usr/local/bin/git
git version 2.25.0
```

## Suggested solution:
I changed `Object.isInvalidated` to be `@objc dynamic` and it resolved the compiler warnings, and the KVO observation still worked:
```diff
- public override final var isInvalidated: Bool { return super.isInvalidated }
+ @objc public dynamic override final var isInvalidated: Bool { return super.isInvalidated }
```

**Comment by ericjordanmossman** [comment]
Thanks for the detail and clear instructions, @sadlerjw.

This is still present in ~>5.0.0, also. I've got to look more into any potential impacts of the suggested change and will get back to you.

**Comment by tgoyne** [comment]
It looks like our tests don't hit this due to the new warning only working if you pass the keypath literal directly to observe(), and our tests happen to have one layer of indirection there.

**Comment by sadlerjw** [comment]
@tgoyne very interesting - thanks for pointing this out, since it allows us to work around the warning until the fix is released, and without disabling warnings-as-errors...just pull the keypath out into a variable and we're good to go!
